Multi-proc build of a .sln file is 'The process cannot access the file because it is being used by another process.'

In the CLI repo, we have a Test.sln file which we `dotnet build`.  However, we are seeing random failures that look like the following:

```
05:25:35.797     1>CSC : error CS2012: Cannot open 'D:\j\workspace\debug_windows---54dcb9ec\test\Msbuild.Tests.Utilities\obj\Debug\netcoreapp1.0\Msbuild.Tests.Utilities.dll' for writing -- 'The process cannot access the file 'D:\j\workspace\debug_windows---54dcb9ec\test\Msbuild.Tests.Utilities\obj\Debug\netcoreapp1.0\Msbuild.Tests.Utilities.dll' because it is being used by another process.' [D:\j\workspace\debug_windows---54dcb9ec\test\Msbuild.Tests.Utilities\Msbuild.Tests.Utilities.csproj] [D:\j\workspace\debug_windows---54dcb9ec\build.proj]
```

See https://ci.dot.net/job/dotnet_cli/job/rel_1.0.0/job/debug_windows_nt_x86_prtest/810/console

To repro:

1. On Windows x64 machine, clone the CLI repo and checkout https://github.com/dotnet/cli/pull/5465/commits/f6b0196b0f8a7342d4b5fe5709568af9e3ede665.  
2. Then remove the line https://github.com/dotnet/cli/pull/5465/commits/f6b0196b0f8a7342d4b5fe5709568af9e3ede665#diff-70eb062c7648e65aa7aff25e924d014bR84 which is working around this problem by turning off multi-proc build.
3. build.cmd

An easier repro once you have built one time is:

0. `F:\cli>git clean test -xdf`
1. `F:\cli>.dotnet_stage0\x64\dotnet.exe restore test\Microsoft.DotNet.Cli.Tests.sln`
2. `F:\cli>.dotnet_stage0\x64\dotnet.exe build test\Microsoft.DotNet.Cli.Tests.sln`


```
F:\cli>.dotnet_stage0\x64\dotnet.exe build test\Microsoft.DotNet.Cli.Tests.sln
Microsoft (R) Build Engine version 15.1.523.56541
Copyright (C) Microsoft Corporation. All rights reserved.

F:\cli\.dotnet_stage0\x64\sdk\2.0.0-alpha-004616\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.DefaultItems.targets(101,5): warning : A PackageReference for 'NETStandard.Library' was included in your project. This package is implicitly referenced by the .NET SDK and you do not typically need to reference it from your project. For more information, see https://aka.ms/sdkimplicitrefs [F:\cli\test\Microsoft.DotNet.Tools.Tests.Utilities\Microsoft.DotNet.Tools.Tests.Utilities.csproj]
F:\cli\.dotnet_stage0\x64\sdk\2.0.0-alpha-004616\Sdks\Microsoft.NET.Sdk\build\Microsoft.NET.Sdk.DefaultItems.targets(101,5): warning : A PackageReference for 'NETStandard.Library' was included in your project. This package is implicitly referenced by the .NET SDK and you do not typically need to reference it from your project. For more information, see https://aka.ms/sdkimplicitrefs [F:\cli\test\Microsoft.DotNet.Tools.Tests.Utilities\Microsoft.DotNet.Tools.Tests.Utilities.csproj]
  ArgumentsReflector -> F:\cli\test\ArgumentsReflector\bin\Debug\netcoreapp2.0\ArgumentsReflector.dll
  Microsoft.DotNet.Cli.Utils -> F:\cli\src\Microsoft.DotNet.Cli.Utils\bin\Debug\netstandard1.5\Microsoft.DotNet.Cli.Utils.dll
  Microsoft.DotNet.TestFramework -> F:\cli\src\Microsoft.DotNet.TestFramework\bin\Debug\netstandard1.5\Microsoft.DotNet.TestFramework.dll
  Microsoft.DotNet.InternalAbstractions -> F:\cli\src\Microsoft.DotNet.InternalAbstractions\bin\Debug\netstandard1.3\Microsoft.DotNet.InternalAbstractions.dll
CSC : error CS2012: Cannot open 'F:\cli\test\Microsoft.DotNet.Tools.Tests.Utilities\obj\Debug\netstandard1.5\Microsoft.DotNet.Tools.Tests.Utilities.dll' for writing -- 'The process cannot access the file 'F:\cli\test\Microsoft.DotNet.Tools.Tests.Utilities\obj\Debug\netstandard1.5\Microsoft.DotNet.Tools.Tests.Utilities.dll' because it is being used by another process.' [F:\cli\test\Microsoft.DotNet.Tools.Tests.Utilities\Microsoft.DotNet.Tools.Tests.Utilities.csproj]
```

I have a `/v:diag` trace of the error.  For any MSFT employee:  `\\scratch2\scratch\eerhardt\multi-proc-build.log`.  For anyone without access, but wants to look at it, I can upload it publicly if you give me a place that takes 250 MB files.

The interesting things to me are these two entries:

```
10:20:51.097  4:16>CoreCompile: (TargetId:274)
                     Task Parameter:NoStandardLib=True (TaskId:145)
                     Task Parameter:Optimize=False (TaskId:145)
                     Task Parameter:Deterministic=True (TaskId:145)
                     Task Parameter:OutputAssembly=obj\Debug\netstandard1.5\Microsoft.DotNet.Tools.Tests.Utilities.dll (TaskId:145)
                     Task Parameter:Prefer32Bit=False (TaskId:145)
```

and 

```
10:20:51.132  4:22>Target "CoreCompile: (TargetId:131)" in file "F:\cli\.dotnet_stage0\x64\sdk\2.0.0-alpha-004622\Roslyn\Microsoft.CSharp.Core.targets" from project "F:\cli\test\Microsoft.DotNet.Tools.Tests.Utilities\Microsoft.DotNet.Tools.Tests.Utilities.csproj" (target "Compile" depends on it):
                   Building target "CoreCompile" completely.
                   Output file "obj\Debug\netstandard1.5\Microsoft.DotNet.Tools.Tests.Utilities.dll" does not exist.
                   Set Property: NoWarn=1701;1702;1705;1701;1702
                   Set Property: UseSharedCompilation=true
                   Using "Csc" task from assembly "F:\cli\.dotnet_stage0\x64\sdk\2.0.0-alpha-004622\Roslyn\Microsoft.Build.Tasks.CodeAnalysis.dll".
                   Task "Csc" (TaskId:70)
                     Task Parameter:AllowUnsafeBlocks=False (TaskId:70)
                     Task Parameter:DebugType=portable (TaskId:70)
                     Task Parameter:DefineConstants=TRACE;NETSTANDARD1_5;DEBUG (TaskId:70)
                     Task Parameter:DisabledWarnings=1701;1702;1705;1701;1702 (TaskId:70)
                     Task Parameter:EmitDebugInformation=True (TaskId:70)
                     Task Parameter:ErrorReport=prompt (TaskId:70)
                     Task Parameter:FileAlignment=512 (TaskId:70)
                     Task Parameter:KeyFile=../../tools/Key.snk (TaskId:70)
                     Task Parameter:NoConfig=True (TaskId:70)
                     Task Parameter:NoLogo=True (TaskId:70)
                     Task Parameter:NoStandardLib=True (TaskId:70)
                     Task Parameter:Optimize=False (TaskId:70)
                     Task Parameter:Deterministic=True (TaskId:70)
                     Task Parameter:OutputAssembly=obj\Debug\netstandard1.5\Microsoft.DotNet.Tools.Tests.Utilities.dll (TaskId:70)
```

and then that last task is failing:

```
                     F:\cli\.dotnet_stage0\x64\sdk\2.0.0-alpha-004622\Roslyn\RunCsc.cmd /noconfig /unsafe- /nowarn:1701,1702,1705,1701,1702 /nostdlib+ /errorreport:prompt /warn:4 /define:TRACE;NETSTANDARD1_5;DEBUG /reference:C:\Users\eerhardt\.nuget\packages\fluentassertions\4.0.0\lib\dotnet\FluentAssertions.Core.dll /reference:C:\Users\eerhardt\.nuget\packages\fluentassertions\4.0.0\lib\dotnet\FluentAssertions.dll /reference:C:\Users\eerhardt\.nuget\packages\microsoft.build\15.1.0-preview-000523-01\lib\netstandard1.5\Microsoft.Build.dll /reference:C:\Users\eerhardt\.nuget\packages\microsoft.build.framework\15.1.0-preview-000523-01\lib\netstandard1.3\Microsoft.Build.Framework.dll /reference:C:\Users\eerhardt\.nuget\packages\microsoft.build.utilities.core\15.1.0-preview-000523-01\lib\netstandard1.3\Microsoft.Build.Utilities.Core.dll /reference:C:\Users\eerhardt\.nuget\packages\microsoft.csharp\4.0.1\ref\netstandard1.0\Microsoft.CSharp.dll /reference:F:\cli\src\Microsoft.DotNet.Cli.Utils\bin\Debug\netstandard1.5\Microsoft.DotNet.Cli.Utils.dll /reference:F:\cli\src\Microsoft.DotNet.InternalAbstractions\bin\Debug\netstandard1.3\Microsoft.DotNet.InternalAbstractions.dll /reference:C:\Users\eerhardt\.nuget\packages\microsoft.dotnet.platformabstractions\1.0.1-beta-000933\lib\netstandard1.3\Microsoft.DotNet.PlatformAbstractions.dll /reference:F:\cli\src\Microsoft.DotNet.TestFramework\bin\Debug\netstandard1.5\Microsoft.DotNet.TestFramework.dll /reference:C:\Users\eerhardt\.nuget\packages\microsoft.extensions.dependencymodel\1.0.1-beta-000933\lib\netstandard1.3\Microsoft.Extensions.DependencyModel.dll /reference:C:\Users\eerhardt\.nuget\packages\microsoft.win32.primitives\4.0.1\ref\netstandard1.3\Microsoft.Win32.Primitives.dll /reference:C:\Users\eerhardt\.nuget\packages\newtonsoft.json\9.0.1\lib\netstandard1.0\Newtonsoft.Json.dll /reference:C:\Users\eerhardt\.nuget\packages\nuget.common\4.0.0-rc3-2193\lib\netstandard1.3\NuGet.Common.dll /reference:C:\Users\eerhardt\.nuget\packages\nuget.configuration\4.0.0-rc3-2193\lib\netstandard1.3\NuGet.Configuration.dll /reference:C:\Users\eerhardt\.nuget\packages\nuget.dependencyresolver.core\4.0.0-rc3-2193\lib\netstandard1.3\NuGet.DependencyResolver.Core.dll /reference:C:\Users\eerhardt\.nuget\packages\nuget.frameworks\4.0.0-rc3-2193\lib\netstandard1.3\NuGet.Frameworks.dll /reference:C:\Users\eerhardt\.nuget\packages\nuget.librarymodel\4.0.0-rc3-2193\lib\netstandard1.3\NuGet.LibraryModel.dll /reference:C:\Users\eerhardt\.nuget\packages\nuget.packaging.core\4.0.0-rc3-2193\lib\netstandard1.3\NuGet.Packaging.Core.dll /reference:C:\Users\eerhardt\.nuget\packages\nuget.packaging.core.types\4.0.0-rc3-2193\lib\netstandard1.3\NuGet.Packaging.Core.Types.dll /reference:C:\Users\eerhardt\.nuget\packages\nuget.packaging\4.0.0-rc3-2193\lib\netstandard1.3\NuGet.Packaging.dll /reference:C:\Users\eerhardt\.nuget\packages\nuget.projectmodel\4.0.0-rc3-2193\lib\netstandard1.3\NuGet.ProjectModel.dll /reference:C:\Users\eerhardt\.nuget\packages\nuget.protocol.core.types\4.0.0-rc3-2193\lib\netstandard1.3\NuGet.Protocol.Core.Types.dll /reference:C:\Users\eerhardt\.nuget\packages\nuget.protocol.core.v3\4.0.0-rc3-2193\lib\netstandard1.3\NuGet.Protocol.Core.v3.dll /reference:C:\Users\eerhardt\.nuget\packages\nuget.repositories\4.0.0-rc3-2193\lib\netstandard1.3\NuGet.Repositories.dll /reference:C:\Users\eerhardt\.nuget\packages\nuget.runtimemodel\4.0.0-rc3-2193\lib\netstandard1.3\NuGet.RuntimeModel.dll /reference:C:\Users\eerhardt\.nuget\packages\nuget.versioning\4.0.0-rc3-2193\lib\netstandard1.0\NuGet.Versioning.dll /reference:C:\Users\eerhardt\.nuget\packages\system.appcontext\4.1.0\ref\netstandard1.3\System.AppContext.dll /reference:C:\Users\eerhardt\.nuget\packages\system.collections.concurrent\4.0.12\ref\netstandard1.3\System.Collections.Concurrent.dll /reference:C:\Users\eerhardt\.nuget\packages\system.collections\4.0.11\ref\netstandard1.3\System.Collections.dll /reference:C:\Users\eerhardt\.nuget\packages\system.collections.immutable\1.2.0\lib\netstandard1.0\System.Collections.Immutable.dll /reference:C:\Users\eerhardt\.nuget\packages\system.collections.nongeneric\4.0.1\ref\netstandard1.3\System.Collections.NonGeneric.dll /reference:C:\Users\eerhardt\.nuget\packages\system.console\4.0.0\ref\netstandard1.3\System.Console.dll /reference:C:\Users\eerhardt\.nuget\packages\system.diagnostics.contracts\4.0.1\ref\netstandard1.0\System.Diagnostics.Contracts.dll /reference:C:\Users\eerhardt\.nuget\packages\system.diagnostics.debug\4.0.11\ref\netstandard1.3\System.Diagnostics.Debug.dll /reference:C:\Users\eerhardt\.nuget\packages\system.diagnostics.fileversioninfo\4.0.0\ref\netstandard1.3\System.Diagnostics.FileVersionInfo.dll /reference:C:\Users\eerhardt\.nuget\packages\system.diagnostics.process\4.1.0\ref\netstandard1.4\System.Diagnostics.Process.dll /reference:C:\Users\eerhardt\.nuget\packages\system.diagnostics.tools\4.0.1\ref\netstandard1.0\System.Diagnostics.Tools.dll /reference:C:\Users\eerhardt\.nuget\packages\system.diagnostics.tracesource\4.0.0\ref\netstandard1.3\System.Diagnostics.TraceSource.dll /reference:C:\Users\eerhardt\.nuget\packages\system.diagnostics.tracing\4.1.0\ref\netstandard1.5\System.Diagnostics.Tracing.dll /reference:C:\Users\eerhardt\.nuget\packages\system.dynamic.runtime\4.0.11\ref\netstandard1.3\System.Dynamic.Runtime.dll /reference:C:\Users\eerhardt\.nuget\packages\system.globalization.calendars\4.0.1\ref\netstandard1.3\System.Globalization.Calendars.dll /reference:C:\Users\eerhardt\.nuget\packages\system.globalization\4.0.11\ref\netstandard1.3\System.Globalization.dll /reference:C:\Users\eerhardt\.nuget\packages\system.io.compression\4.1.0\ref\netstandard1.3\System.IO.Compression.dll /reference:C:\Users\eerhardt\.nuget\packages\system.io.compression.zipfile\4.0.1\ref\netstandard1.3\System.IO.Compression.ZipFile.dll /reference:C:\Users\eerhardt\.nuget\packages\system.io\4.1.0\ref\netstandard1.5\System.IO.dll /reference:C:\Users\eerhardt\.nuget\packages\system.io.filesystem\4.0.1\ref\netstandard1.3\System.IO.FileSystem.dll /reference:C:\Users\eerhardt\.nuget\packages\system.io.filesystem.primitives\4.0.1\ref\netstandard1.3\System.IO.FileSystem.Primitives.dll /reference:C:\Users\eerhardt\.nuget\packages\system.io.pipes\4.0.0\ref\netstandard1.3\System.IO.Pipes.dll /reference:C:\Users\eerhardt\.nuget\packages\system.linq\4.1.0\ref\netstandard1.0\System.Linq.dll /reference:C:\Users\eerhardt\.nuget\packages\system.linq.expressions\4.1.0\ref\netstandard1.3\System.Linq.Expressions.dll /reference:C:\Users\eerhardt\.nuget\packages\system.linq.parallel\4.0.1\ref\netstandard1.1\System.Linq.Parallel.dll /reference:C:\Users\eerhardt\.nuget\packages\system.net.http\4.1.0\ref\netstandard1.3\System.Net.Http.dll /reference:C:\Users\eerhardt\.nuget\packages\system.net.primitives\4.0.11\ref\netstandard1.3\System.Net.Primitives.dll /reference:C:\Users\eerhardt\.nuget\packages\system.net.sockets\4.1.0\ref\netstandard1.3\System.Net.Sockets.dll /reference:C:\Users\eerhardt\.nuget\packages\system.objectmodel\4.0.12\ref\netstandard1.3\System.ObjectModel.dll /reference:C:\Users\eerhardt\.nuget\packages\system.reflection\4.1.0\ref\netstandard1.5\System.Reflection.dll /reference:C:\Users\eerhardt\.nuget\packages\system.reflection.extensions\4.0.1\ref\netstandard1.0\System.Reflection.Extensions.dll /reference:C:\Users\eerhardt\.nuget\packages\system.reflection.metadata\1.3.0\lib\netstandard1.1\System.Reflection.Metadata.dll /reference:C:\Users\eerhardt\.nuget\packages\system.reflection.primitives\4.0.1\ref\netstandard1.0\System.Reflection.Primitives.dll /reference:C:\Users\eerhardt\.nuget\packages\system.reflection.typeextensions\4.1.0\ref\netstandard1.5\System.Reflection.TypeExtensions.dll /reference:C:\Users\eerhardt\.nuget\packages\system.resources.reader\4.0.0\lib\netstandard1.0\System.Resources.Reader.dll /reference:C:\Users\eerhardt\.nuget\packages\system.resources.resourcemanager\4.0.1\ref\netstandard1.0\System.Resources.ResourceManager.dll /reference:C:\Users\eerhardt\.nuget\packages\system.runtime\4.1.0\ref\netstandard1.5\System.Runtime.dll /reference:C:\Users\eerhardt\.nuget\packages\system.runtime.extensions\4.1.0\ref\netstandard1.5\System.Runtime.Extensions.dll /reference:C:\Users\eerhardt\.nuget\packages\system.runtime.handles\4.0.1\ref\netstandard1.3\System.Runtime.Handles.dll /reference:C:\Users\eerhardt\.nuget\packages\system.runtime.interopservices\4.1.0\ref\netstandard1.5\System.Runtime.InteropServices.dll /reference:C:\Users\eerhardt\.nuget\packages\system.runtime.interopservices.runtimeinformation\4.0.0\ref\netstandard1.1\System.Runtime.InteropServices.RuntimeInformation.dll /reference:C:\Users\eerhardt\.nuget\packages\system.runtime.loader\4.0.0\ref\netstandard1.5\System.Runtime.Loader.dll /reference:C:\Users\eerhardt\.nuget\packages\system.runtime.numerics\4.0.1\ref\netstandard1.1\System.Runtime.Numerics.dll /reference:C:\Users\eerhardt\.nuget\packages\system.runtime.serialization.primitives\4.1.1\ref\netstandard1.3\System.Runtime.Serialization.Primitives.dll /reference:C:\Users\eerhardt\.nuget\packages\system.runtime.serialization.xml\4.1.1\ref\netstandard1.3\System.Runtime.Serialization.Xml.dll /reference:C:\Users\eerhardt\.nuget\packages\system.security.cryptography.algorithms\4.2.0\ref\netstandard1.4\System.Security.Cryptography.Algorithms.dll /reference:C:\Users\eerhardt\.nuget\packages\system.security.cryptography.encoding\4.0.0\ref\netstandard1.3\System.Security.Cryptography.Encoding.dll /reference:C:\Users\eerhardt\.nuget\packages\system.security.cryptography.primitives\4.0.0\ref\netstandard1.3\System.Security.Cryptography.Primitives.dll /reference:C:\Users\eerhardt\.nuget\packages\system.security.cryptography.protecteddata\4.0.0\ref\netstandard1.3\System.Security.Cryptography.ProtectedData.dll /reference:C:\Users\eerhardt\.nuget\packages\system.security.cryptography.x509certificates\4.1.0\ref\netstandard1.4\System.Security.Cryptography.X509Certificates.dll /reference:C:\Users\eerhardt\.nuget\packages\system.security.principal\4.0.1\ref\netstandard1.0\System.Security.Principal.dll /reference:C:\Users\eerhardt\.nuget\packages\system.text.encoding\4.0.11\ref\netstandard1.3\System.Text.Encoding.dll /reference:C:\Users\eerhardt\.nuget\packages\system.text.encoding.extensions\4.0.11\ref\netstandard1.3\System.Text.Encoding.Extensions.dll /reference:C:\Users\eerhardt\.nuget\packages\system.text.regularexpressions\4.1.0\ref\netstandard1.3\System.Text.RegularExpressions.dll /reference:C:\Users\eerhardt\.nuget\packages\system.threading\4.0.11\ref\netstandard1.3\System.Threading.dll /reference:C:\Users\eerhardt\.nuget\packages\system.threading.tasks.dataflow\4.6.0\lib\netstandard1.1\System.Threading.Tasks.Dataflow.dll /reference:C:\Users\eerhardt\.nuget\packages\system.threading.tasks\4.0.11\ref\netstandard1.3\System.Threading.Tasks.dll /reference:C:\Users\eerhardt\.nuget\packages\system.threading.thread\4.0.0\ref\netstandard1.3\System.Threading.Thread.dll /reference:C:\Users\eerhardt\.nuget\packages\system.threading.threadpool\4.0.10\ref\netstandard1.3\System.Threading.ThreadPool.dll /reference:C:\Users\eerhardt\.nuget\packages\system.threading.timer\4.0.1\ref\netstandard1.2\System.Threading.Timer.dll /reference:C:\Users\eerhardt\.nuget\packages\system.xml.readerwriter\4.0.11\ref\netstandard1.3\System.Xml.ReaderWriter.dll /reference:C:\Users\eerhardt\.nuget\packages\system.xml.xdocument\4.0.11\ref\netstandard1.3\System.Xml.XDocument.dll /reference:C:\Users\eerhardt\.nuget\packages\system.xml.xmldocument\4.0.1\ref\netstandard1.3\System.Xml.XmlDocument.dll /reference:C:\Users\eerhardt\.nuget\packages\system.xml.xpath\4.0.1\ref\netstandard1.3\System.Xml.XPath.dll /reference:C:\Users\eerhardt\.nuget\packages\system.xml.xpath.xmldocument\4.0.1\ref\netstandard1.3\System.Xml.XPath.XmlDocument.dll /reference:C:\Users\eerhardt\.nuget\packages\xunit.abstractions\2.0.1\lib\netstandard1.0\xunit.abstractions.dll /reference:C:\Users\eerhardt\.nuget\packages\xunit.assert\2.2.0-beta4-build3444\lib\netstandard1.0\xunit.assert.dll /reference:C:\Users\eerhardt\.nuget\packages\xunit.extensibility.core\2.2.0-beta4-build3444\lib\netstandard1.0\xunit.core.dll /reference:C:\Users\eerhardt\.nuget\packages\xunit.extensibility.execution\2.2.0-beta4-build3444\lib\netstandard1.0\xunit.execution.dotnet.dll /debug+ /debug:portable /filealign:512 /keyfile:../../tools/Key.snk /nologo /optimize- /out:obj\Debug\netstandard1.5\Microsoft.DotNet.Tools.Tests.Utilities.dll /target:library /warnaserror- /utf8output /deterministic+ Assertions\CommandResultAssertions.cs Assertions\CommandResultExtensions.cs Assertions\DirectoryInfoAssertions.cs Assertions\DirectoryInfoExtensions.cs Assertions\FileInfoAssertions.cs Assertions\FileInfoExtensions.cs Assertions\FileInfoExtensions.FileInfoLock.cs Assertions\FileInfoExtensions.FileInfoNuGetLock.cs Assertions\StringAssertionsExtensions.cs Commands\AddReferenceCommand.cs Commands\BuildCommand.cs Commands\BuildPJCommand.cs Commands\CleanCommand.cs Commands\DependencyToolInvokerCommand.cs Commands\DotnetCommand.cs Commands\DotnetTestCommand.cs Commands\HelpCommand.cs Commands\ListReferenceCommand.cs Commands\MigrateCommand.cs Commands\MSBuildCommand.cs Commands\NewCommand.cs Commands\PackCommand.cs Commands\PublishCommand.cs Commands\RemoveReferenceCommand.cs Commands\RestoreCommand.cs Commands\RestoreProjectJsonCommand.cs Commands\RunCommand.cs Commands\TestCommand.cs Commands\VSTestCommand.cs DotnetLegacyRuntimeIdentifiers.cs Extensions\IDisposableExtensions.cs Extensions\ProcessExtensions.cs Extensions\TestCommandExtensions.cs Mock\EnvironmentMockBuilder.cs Mock\FileSystemMockBuilder.cs Mock\ITemporaryDirectoryMock.cs NetworkUtils\NetworkHelper.cs NetworkUtils\PortManager.cs NuGetConfig.cs PeReaderUtils.cs ProjectUtils.cs Properties\Properties.cs RepoDirectoriesProvider.cs TempFileSystem\DisposableDirectory.cs TempFileSystem\DisposableFile.cs TempFileSystem\FileNameUtilities.cs TempFileSystem\ImmutableArrayTestExtensions.cs TempFileSystem\PathKind.cs TempFileSystem\PathUtilities.cs TempFileSystem\TempDirectory.cs TempFileSystem\TempFile.cs TempFileSystem\TempRoot.cs TestBase.cs WindowsOnlyFactAttribute.cs "C:\Users\eerhardt\AppData\Local\Temp\.NETStandard,Version=v1.5.AssemblyAttributes.cs" obj\Debug\netstandard1.5\Microsoft.DotNet.Tools.Tests.Utilities.AssemblyInfo.cs (TaskId:70)
10:20:52.906  4:22>CSC : error CS2012: Cannot open 'F:\cli\test\Microsoft.DotNet.Tools.Tests.Utilities\obj\Debug\netstandard1.5\Microsoft.DotNet.Tools.Tests.Utilities.dll' for writing -- 'The process cannot access the file 'F:\cli\test\Microsoft.DotNet.Tools.Tests.Utilities\obj\Debug\netstandard1.5\Microsoft.DotNet.Tools.Tests.Utilities.dll' because it is being used by another process.' [F:\cli\test\Microsoft.DotNet.Tools.Tests.Utilities\Microsoft.DotNet.Tools.Tests.Utilities.csproj]
                     Output Item(s): CscCommandLineArgs= (TaskId:70)
                   Done executing task "Csc" -- FAILED. (TaskId:70)
```

As you can see, node 16 and node 22 are trying to compile the same project to the same output file roughly 35ms apart.

@rainersigwald @jeffkl @cdmihai - doesn't MSBuild have a locking mechanism so 2 nodes don't try building the same project at the same time?

--- a/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.Common.targets
+++ b/src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Sdk.Common.targets
@@ -55,13 +55,17 @@ Copyright (c) .NET Foundation. All rights reserved.
     with the referencing project's target framework.
   ============================================================
    -->
-  <Target Name="GetTargetFrameworkProperties" Returns="TargetFramework=$(NearestTargetFramework)">
+  <Target Name="GetTargetFrameworkProperties" Returns="TargetFramework=$(NearestTargetFramework);ProjectHasSingleTargetFramework=$(_HasSingleTargetFramework)">
 
     <PropertyGroup>
       <!-- If a ReferringTargetFramework was not specified, and we only have one TargetFramework, then don't try to check compatibility -->
       <_SkipNearestTargetFrameworkResolution Condition="'$(TargetFramework)' != '' and '$(ReferringTargetFramework)' == ''">true</_SkipNearestTargetFrameworkResolution>
       <NearestTargetFramework Condition="'$(_SkipNearestTargetFrameworkResolution)' == 'true'">$(TargetFramework)</NearestTargetFramework>
-      
+
+      <!-- A project can only have more than one output if the current global properties are such that the current build is a cross-targeting one. -->
+      <_HasSingleTargetFramework Condition="'$(IsCrossTargetingBuild)' != 'true'">true</_HasSingleTargetFramework>
+      <_HasSingleTargetFramework Condition="'$(_HasSingleTargetFramework)' == ''">false</_HasSingleTargetFramework>
+
       <_PossibleTargetFrameworks Condition="'$(TargetFramework)' != ''">$(TargetFramework)</_PossibleTargetFrameworks>
       <_PossibleTargetFrameworks Condition="'$(TargetFramework)' == ''">$(TargetFrameworks)</_PossibleTargetFrameworks>
     </PropertyGroup>

